### ISSUE TITLE
Pre-canned set of filters for cleaning – 3 to 4 selected facets (TBD) with default parameters than can be changed (pop-up window)

### ISSUE THREAD

**Issue by sat01a** [AtlasOfLivingAustralia/spatial-hub#191]
Belongs to Data filtering in the SP #180

**Comment by Tasilee** [comment]
It isn't obvious to me what 'CSDM default filter' is in the dropdown facet box. I'd suggest

1. Calling it 'CSDM filter workflow...' to differentiate it from facets and align with 'Search facets...'
1. Rename pop-up window title to 'CSDM filter workflow' is important for reinforcement that this is a special workflow and does not operate strictly like the 'standard' workflow.
1. There is no EDIT button so I presume you can't edit other than skipping steps?
1. The first annotation looks like a real one, but why can I type in text if it can't be saved? Not good.
1. The step 'annotations' are more 'Actions' 
1. A spin-off of the CSDM work may well be a subset of facet filters that users may want to apply. This is guess relates back to my comment on #194 about exporting and importing workflows. It may be neat to have a small repository of 'published' workflows that demonstrate a process or a case study etc.

**Comment by CorinnaPaeper** [comment]
Agreed on pre-canned filters as per the endorsed recommendations of the Expert Committee: ([Document](https://docs.google.com/document/d/1Q67uUHS8ZC3HxeNC0Wr64dLWZRgoX5rtT0btB9F3JAg/edit#)):

1. Year range
2. Remove duplicate identical coordinates
3. Remove records with any of the following issues: name not supplied; name not recognised; supplied coordinates are zero; zero latitude or longitude; decimal latitude/longitude conversion failed; cultivated/escapee; unparseable verbatim coordinates; coordinates are centre of country or centre of state; unable to convert UTM coordinates.

"3" refers to information provided in the [initial discussion document](https://docs.google.com/document/d/17LKYENI03IsQs6sXVfTxIlzSe7i8sHaNqWE4KwJgMRs/edit#) by Kristen Williams – see relevant excerpt below. This option should be basically an on/off switch that allows the user to apply all of these filters in one hit (or not)… I think when we spoke about this a while ago, all of the quality assertions made sense, apart from the last one. I don’t think anyone was able to identify which QA assertion this statement belongs to. The full script and additional package information are available here: https://github.com/cwarecsiro/gdmEngine/tree/master/gdmEngine - https://github.com/cwarecsiro/gdmEngine/blob/master/gdmEngine/R/filter_ALA_data.R

> "Then, a larger series of filters are applied with the function, including standardising the output to the level of species. These filters include ALA quality assertions, all of which could easily be omitted (or additional assertions added). Here’s most of them (where the quality assertion is ‘true’ the data record is removed from the dataset to be used in a model):
>  
> ·         Name.not.supplied=='true'
> ·         Name.not.recognised=='true'
> ·         Cultivated...escapee=='true'
> ·         Unparseable.verbatim.coordinates=='true'
> ·         Decimal.latitude.longitude.conversion.failed=='true'
> ·         Unable.to.convert.UTM.coordinates=='true'
> ·         Supplied.coordinates.are.zero=='true'
> ·         Zero.latitude=='true'
> ·         Zero.longitude=='true'
> ·         Coordinates.centre.of.country=='true'
> ·         Supplied.coordinates.centre.of.state=='true'
> ·         processing.Error=='true'"

**Comment by m-hope** [comment]
As a starting point, the default CSDM filter set should be:
- year > 1970
- no duplicate coordinates
- spatially valid

**Comment by Tasilee** [comment]
Sounds reasonable as a starting point and I'm chiming in here as I have had a gripe about "Spatially valid", most recently from my January 2020 report (but raised aeons ago). I would create an issue for it, but I'd better leave that to Nick as to where it would go.

"We don't have an easily accessible definition of "Spatially valid" so I figured this needed to be addressed and amended. The definitions of the tests that the ALA runs can be found https://biocache-ws.ala.org.au/ws/assertions/codes.  The current formal definition of "Spatially valid" is isFatal: true and code < 10000, that is the following tests/flags

1.	Supplied coordinates are zero 
2.	Suspected outlier 
3.	Unable to convert UTM coordinates 
4.	Zero latitude 
5.	Unparseable verbatim coordinates 
6.	Coordinates centre of country 
7.	Geospatial issue ?? No idea what this means
8.	Coordinates are out of range for species ?? No idea what this means
9.	Outside expert range for species 
10.	Supplied coordinates centre of state 
11.	Decimal latitude/longitude conversion failed 
12.	Zero longitude 
13.	Habitat incorrect for species (presume marine species on land or vice-versa)

I would not have included **2, 6, 8, 9 or 10 as FATAL**. This I have raised with Dave Martin (who coded this originally)." 

....but no response so far, so trying a different route.
